sql/src/main/java/io/crate/cluster/gracefulstop/DecommissioningService.java

```diff
@@ -277,6 +253,36 @@ void exitIfNoActiveRequests(final long startTime) {
         executorService.schedule(() -> exitIfNoActiveRequests(startTime), 5, TimeUnit.SECONDS);
     }
 
+    private CompletableFuture<ClusterUpdateSettingsResponse> clusterUpdateSettings(final Settings settings) {
+        /*
+         * setting this setting will cause the {@link DecommissionAllocationDecider} to prevent allocations onto this node
+         *
+         * nodeIds are part of the key to prevent conflicts if other nodes are being decommissioned in parallel
+         */
+        FutureActionListener<ClusterUpdateSettingsResponse, ClusterUpdateSettingsResponse> settingsResponseFutureListener
```

Comment:
is there a better way to obtain a `CompletableFuture` than using the `FutureActionListener`? It looks rather strange for this case.

Reply:
I don't think so